app/src/main/java/com/github/factotum_sdp/factotum/data/LoginDataSource.kt

```diff
@@ -0,0 +1,24 @@
+package com.github.factotum_sdp.factotum.data
+
+import com.github.factotum_sdp.factotum.data.model.LoggedInUser
+import java.io.IOException
+
+/**
+ * Class that handles authentication w/ login credentials and retrieves user information.
+ */
+class LoginDataSource {
+
+    fun login(username: String, password: String): Result<LoggedInUser> {
+        try {
+            // TODO: handle loggedInUser authentication
+            val fakeUser = LoggedInUser(java.util.UUID.randomUUID().toString(), username)
+            return Result.Success(fakeUser)
+        } catch (e: Throwable) {
+            return Result.Error(IOException("Error logging in", e))
+        }
+    }
+
+    fun logout() {
+        // TODO: revoke authentication
```

Comment:
Logout and revoking authentification seems like 2 different processes, which one are you trying to implement here ?

Reply:
We are probably going to use Firebase's methods to logout.
These methods probaly revoke the authentication to log you out